utils/workers/Creators.py

```diff
@@ -0,0 +1,364 @@
+
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from dataclasses import dataclass
+import glob
+import logging
+import numpy as np
+from pathlib import Path
+import pydicom
+import re
+from scipy import stats
+
+from utils.objects.Labels import RoiLabel
+import utils.roi_tools as roi_tools
+from utils.workers import Converters
+
+log = logging.getLogger(__name__)
+
+"""
+Generators - 
+Process 3 of 4
+
+This process actually generates the ROI image.  Due to some functionality that 
+this code needs to have, the generator will contain a converter.  Therefore it
+is the generators responsibility to not only create the ROI images but also
+CALL the converter. 
+
+Responsibilities:
+1. based on user input (binary, bitmask, combine, etc), generate images of the ROI
+2. generate the "labels" object to track mask size and bit value, etc.
+3. generate the name of the new file to be saved
+2. call the converter
+
+
+Full process:
+1. Prep
+2. Collect
+3. Generate
+4. Convert
+
+"""
+
+
+class CreateWorker(ABC):
+    def __init__(self, orig_dir, roi_dir, output_dir, base_file_name, combine, bitmask, converter):
+        self.orig_dir = orig_dir
+        self.roi_dir = roi_dir
+        self.output_dir = output_dir
+        self.base_file_name = base_file_name
+        self.combine = combine
+        self.bitmask = bitmask
+        self.converter = converter
+
+        ## End of user defined properties
+        self.shape = [0, 0, 0]
+        self.max_labels = 0
+        self.dtype = np.uint8
+        self.bits = 8
+
+    def get_labels(self, ohifviewer_info):
+        """
+        gather_ROI_info extracts label-name along with bitmasked index and RGBA
+            color code for each distinct label in the ROI collection.
+
+        Args:
+            file_obj (flywheel.models.file.File): The flywheel file-object with
+                ROI data contained within the `.info.ROI` metadata.
+
+        Returns:
+            OrderedDict: the label object populated with ROI attributes
+        """
+
+        # dictionary for labels, index, R, G, B, A
+        labels = OrderedDict()
+
+        # React OHIF Viewer
+        roi_list = [
+            individual_roi
+            for roi_type_list in ohifviewer_info.values()
+            for individual_roi in roi_type_list
+        ]
+
+        for roi in roi_list:
+            if roi.get("location"):
+                if roi["location"] not in labels.keys():
+                    label_index = int(2 ** (len(labels)))
+                    labels[roi["location"]] = RoiLabel(
+                        label=roi["location"],
+                        index=label_index,
+                        color="fbbc05",
```

Comment:
Whats this about?

Reply:
a way to set/store roi color by hex.  It's a bit convoluted but it's used in "write_3D_Slicer_CTBL".  When we made this at first flywheel didn't have colors, so we just picked one and tried to cover our bases with both hex and RGB.  Now that color is implemented, I could and should update this.